src/components/node-list/node-list-group.js

```diff
@@ -51,14 +55,77 @@ export const NodeListGroup = ({
         />
       </NodeListRow>
     </h3>
-
-    <ChildrenContainer
-      className={classnames(childrenClassName, 'pipeline-nodelist__children', {
-        'pipeline-nodelist__children--closed': collapsed
-      })}>
-      {children}
-    </ChildrenContainer>
-  </Container>
+    <LazyList
+      name={name}
+      height={(start, end) => (end - start) * nodeListRowHeight}
+      total={items.length}
+      onChange={({ start, end, total }) =>
+        console.log(
+          `${group.name} ${start} to ${end} (${end - start} of ${total})`
+        )
+      }>
+      {({
+        start,
+        end,
+        total,
+        listRef,
+        upperRef,
+        lowerRef,
+        listStyle,
+        upperStyle,
+        lowerStyle
+      }) => (
+        <ul
+          ref={listRef}
+          style={listStyle}
+          className={modifiers(
+            'pipeline-nodelist__children',
+            { closed: collapsed },
+            'pipeline-nodelist__list'
+          )}>
+          <li
+            className={modifiers('pipeline-nodelist__placeholder-upper', {
+              fade: start > 0
+            })}
+            ref={upperRef}
+            style={upperStyle}
+          />
+          <li
+            className={modifiers('pipeline-nodelist__placeholder-lower', {
+              fade: end < total
+            })}
+            ref={lowerRef}
+            style={lowerStyle}
+          />
+          {items.slice(start, end).map(item => (
+            <NodeListRow
+              container="li"
+              key={item.id}
+              id={item.id}
+              kind={group.kind}
+              label={item.highlightedLabel}
+              name={item.name}
+              type={item.type}
+              active={item.active}
+              checked={item.checked}
+              disabled={item.disabled}
+              faded={item.faded}
+              visible={item.visible}
+              selected={item.selected}
+              unset={item.unset}
+              visibleIcon={item.visibleIcon}
+              invisibleIcon={item.invisibleIcon}
+              onClick={() => onItemClick(item)}
+              // Disabled to avoid unrelated hover lag for now.
+              // onMouseEnter={() => onItemMouseEnter(item)}
+              // onMouseLeave={() => onItemMouseLeave(item)}
```

Comment:
`// @TODO fix this`?

Reply:
Sure I'll update this part once I figure out next steps.

Reply:
I have added a flag and enabled hover highlighting again, but without the linked nodes search when the flag is enabled. That should work well enough until a non-blocking solution for linked nodes is implemented.